client/network/common/src/sync/message.rs

```diff
@@ -218,3 +220,27 @@ pub mod generic {
 		}
 	}
 }
+
+/// Handshake sent when we open a block announces substream.
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+pub struct BlockAnnouncesHandshake<B: BlockT> {
```

Comment:
Can this not directly life in the sync crate?

Reply:
Not right now no because each new imported block causes the code in `protocol.rs` to update the handshake code. I'll move it to `sc-network-sync` when I'm fixing that code.